src/dataframe/dataframe.jl

```diff
@@ -262,8 +264,15 @@ DataFrame(columns::NTuple{N, AbstractVector}; copycols::Bool=true) where {N} =
     DataFrame(collect(AbstractVector, columns), gennames(length(columns)),
               copycols=copycols)
 
-DataFrame(columns::AbstractMatrix,
-          cnames::AbstractVector{Symbol} = gennames(size(columns, 2));
+DataFrame(columns::T) where {T <: AbstractMatrix} =
+    if Tables.istable(T)
+        fromcolumns(Tables.columns(x), collect(Symbol, Tables.columnnames(cols)),
```

Comment:
```suggestion
        fromcolumns(Tables.columns(columns),
                    collect(Symbol, Tables.columnnames(columns)),
```

Reply:
☝️ Hit this trying to test out the branch